MapboxNavigation/RouteTableViewController.swift

```diff
@@ -68,20 +68,43 @@ class RouteTableViewController: UIViewController {
 extension RouteTableViewController: UITableViewDelegate, UITableViewDataSource {
 
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return routeController.routeProgress.route.legs.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        // Don't display section header if there is only one step
+        guard routeController.routeProgress.route.legs.count > 1 else {
+            return nil
+        }
+        let leg = routeController.routeProgress.route.legs[section]
+        
+        let sourceName = leg.source.name
+        let destinationName = leg.destination.name
+        
+        if let sourceName = sourceName?.nonEmptyString, let destinationName = destinationName?.nonEmptyString {
+            return "\(sourceName) to \(destinationName)"
+        }
+        
+        let legToFrom = leg.name.components(separatedBy: ", ")
+        if legToFrom.count == 2 {
+            return "\(legToFrom[0]) to \(legToFrom[1])"
```

Comment:
Not sure why GitHub thinks https://github.com/mapbox/mapbox-navigation-ios/pull/270#discussion_r123045968 is outdated, but it’s still current.